src/gosb/backend.go

```diff
@@ -13,6 +15,9 @@ type backendConfig struct {
 	//Functions for hooks in the runtime
 	transfer func(oldid, newid int, start, size uintptr)
 	register func(id int, start, size uintptr)
+	execute  func(id c.SandId)
```

Comment:
Hook executed when the goroutine is scheduled. Based on id, it should either enable or disable protection mechanisms.